admin/storage_purge

```diff
@@ -102,14 +93,10 @@ function deleteFromS3() {
 		// See if file was marked as used recently in the database. The modification time in S3
 		// isn't updated when an existing file is reused, so this is common.
 		$info = new Zotero_StorageFileInfo;
-		try {
-			$info->parseFromS3Key($key);
-		}
-		// Ignore other files ('bookmarklet_upload.html')
-		catch (Exception $e) {
-			continue;
-		}
 		
+		$info->filename = $key;
+		//$info->zip = ?;
+        
 		$info = Zotero_Storage::getLocalFileInfo($info);
```

Comment:
There is a short window between getting file info and deleting it, where it can be updated by other user and result in a missing  S3 file. One option would be to move this code in transaction.